### DISCUSSION TITLE
Error: Type 'UnmodifiableUint8ListView' not found

### DISCUSSION THREAD

**Discussion by Eferigho** [codemagic-ci-cd/codemagic-docs#2697] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. I click on Start Build Button
2. Preparing build machine: successful

3. Fetching app sources: successful

4. Installing SDKs: successful

5. Setup the keystore: successful

6. Setup App Environment Variables: successful
7. Build AAB : Failed

### Expected results

Build successful

### Actual results

../.pub-cache/hosted/pub.dev/win32-5.0.6/lib/src/guid.dart:32:9: Error: Type 'UnmodifiableUint8ListView' not found.
  final UnmodifiableUint8ListView bytes;
        ^^^^^^^^^^^^^^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/win32-5.0.6/lib/src/guid.dart:32:9: Error: 'UnmodifiableUint8ListView' isn't a type.
  final UnmodifiableUint8ListView bytes;
        ^^^^^^^^^^^^^^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/win32-5.0.6/lib/src/guid.dart:52:17: Error: Method not found: 'UnmodifiableUint8ListView'.
    return Guid(UnmodifiableUint8ListView(guid));
                ^^^^^^^^^^^^^^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/win32-5.0.6/lib/src/guid.dart:56:31: Error: Method not found: 'UnmodifiableUint8ListView'.
  factory Guid.zero() => Guid(UnmodifiableUint8ListView(Uint8List(16)));
                              ^^^^^^^^^^^^^^^^^^^^^^^^^
../.pub-cache/hosted/pub.dev/win32-5.0.6/lib/src/guid.dart:100:17: Error: Method not found: 'UnmodifiableUint8ListView'.
    return Guid(UnmodifiableUint8ListView(Uint8List.fromList(guidAsBytes)));
                ^^^^^^^^^^^^^^^^^^^^^^^^^
Target kernel_snapshot_program failed: Exception

FAILURE: Build failed with an exception.

* Where:
Script '/Users/builder/programs/flutter/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy' line: 1687

* What went wrong:
Execution failed for task ':app:compileFlutterBuildRelease'.
> Process 'command '/Users/builder/programs/flutter/bin/flutter'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 56s
Running Gradle task 'bundleRelease'...                            117.3s
Gradle task bundleRelease failed with exit code 1

Build failed :|
Step 5 script `Build AAB` exited with status code 1

### Build id (optional)

66cecf7c714ca5bb469d15a4

**Comment by himesh-cm** [comment]
Hey @Eferigho 

The error is with the plugin win32,  as shown below.

```
/win32-5.0.6/lib/src/guid.dart:32:9: Error: 'UnmodifiableUint8ListView' isn't a type.
  final UnmodifiableUint8ListView bytes;
```
You should be able to reproduce this issue locally as well. Currently, on codemagic, you are using Flutter version `3.24.1`. What version do you use locally? Please ensure that your development and codemagic versions are the same.

You can modify the `flutter: stable` to the required version in your `yaml`.

If you use the same version, please try it with the latest version of`win32` plugin.

**Comment by himesh-cm** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
